apps/frontend/src/app/2_molecules/AmountRenderer/AmountRenderer.tsx

```diff
@@ -98,6 +79,55 @@ export const AmountRenderer: FC<AmountRendererProps> = ({
     [useTooltip, valueIsRounded],
   );
 
+  const calculateDecimalPlaces = useCallback(
+    (value: Decimalish, precision: number) => {
+      const decimalValue = decimalic(value).toString();
+      const [, decimals = ''] = decimalValue.split('.');
+      const nonZeroIndex = decimals.search(/[^0]/);
+      return nonZeroIndex !== -1
+        ? Math.max(nonZeroIndex + 1, precision)
+        : precision;
+    },
+    [],
+  );
+
+  const calculatedPrecision = useMemo(
+    () => calculateDecimalPlaces(value, precision),
+    [value, precision, calculateDecimalPlaces],
+  );
+
+  const isValueBetweenZeroAndOne = useCallback((value: number) => {
+    const isGreaterThanZero = value > 0;
+    const isLessThanOne = value < 1;
+    return isGreaterThanZero && isLessThanOne;
+  }, []);
```

Comment:
`const isValueBetweenZeroAndOne = useCallback((value: number) => value > 0 && value < 1, []);`

Reply:
i think this function should be moved to a util file